pkg/runner/runner.go

```diff
@@ -126,12 +135,27 @@ func Run(cfg *Config) error {
 		}
 	}
 
-	router, err := api.GenerateRouter(&cfg.APICfg, client, elector, client.TelemetryEngine)
+	router, err := api.GenerateRouter(&cfg.APICfg, client, elector)
 	if err != nil {
 		log.Error("msg", "aborting startup due to error", "err", fmt.Sprintf("generate router: %s", err.Error()))
 		return fmt.Errorf("generate router: %w", err)
 	}
 
+	var telemetryEngine telemetry.Engine
+	engine, err := telemetry.NewEngine(client.Connection, PromscaleID, client.Queryable())
+	if err != nil {
+		log.Debug("msg", "err creating telemetry engine", "err", err.Error())
+	}
+	if engine == nil {
+		telemetryEngine = telemetry.NewNoopEngine()
+	} else {
+		telemetryEngine = engine
+	}
+	telemetryEngine.Start()
+	defer telemetryEngine.Stop()
+
+	registerMetricsForTelemetry(telemetryEngine)
```

Comment:
With removal of line 144:
```suggestion
	if engine == nil {
		engine = telemetry.NewNoopEngine()
	} 
	engine.Start()
	defer engine.Stop()

	registerMetricsForTelemetry(engine)
```